tests/datastore/test_base.py

```diff
@@ -214,3 +216,47 @@ def test_schema_to_store(schemas, expected_class, expected):
     with expected:
         stores = [schema_to_store(schema) for schema in schemas]
         assert all(store == expected_class for store in stores)
+
+
+@pytest.mark.parametrize(
+    "additional_filters, message",
+    [
+        ([("x", "=", 3)], ""),
+        (
+            [[("x", "=", 3), ("x", "=", 4), ("x", "=", 5)]],
+            "mlrun supports additional_filters only as a list of tuples.",
+        ),
+        (
+            [[("x", "=", 3), ("x", "=", 4)]],
+            "mlrun supports additional_filters only as a list of tuples.",
+        ),
+        (("x", "=", 3), "mlrun supports additional_filters only as a list of tuples."),
+        ([("x", "in", [3, 4]), ("y", "in", [3, 4])], ""),
+        (
+            [("age", "=", float("nan"))],
+            "using NaN in additional_filters is not supported",
+        ),
+        (
+            [("age", "in", [10, float("nan")])],
+            "using NaN in additional_filters is not supported",
+        ),
+        ([("x", "=", "=", 3), ("y", "in", [3, 4])], "illegal filter tuple length"),
+        ([()], ""),
+    ],
+)
+def test_revert_list_filters_to_tuple(additional_filters, message):
```

Comment:
If the method being tested is in utils (which maybe it shouldn't be), then the test should be in `test_utils.py`.